index.js

```diff
@@ -23,6 +24,14 @@ function isValidRewritePaths(rewritePaths) {
 	});
 }
 
+function isValidPrecompile(precompile) {
+	return typeof precompile === 'boolean' || precompile === undefined;
+}
+
+async function compileTypeScript(projectDir) {
+	return execa.command(`npx tsc --build ${projectDir}/../tsconfig.json`);
```

Comment:
I don't think that's a good way to resolve `tsconfig.json`. The tree may look like this:

```
- code
-- source
--- index.js
-- tests
--- test.js
tsconfig.json
package.json
```

Reply:
Would this work with `execa.node()` and the `preferLocal` option enabled?

We should assume the `tsconfig.json` is inside the `projectDir`, which is the directory that has the `package.json`. Making the location configurable can be a follow-up feature.

We should opt-in to incremental compilation.

What directory does this write to? Is this based on the configuration? It might be good to override that — if you have AVA watch mode running, and your own `tsc -w` it may lead to trouble if both are updating the same output directory.